process/logsevents/delegatorsProcessor.go

```diff
@@ -68,12 +68,17 @@ func (dp *delegatorsProc) processEvent(args *argsProcessEvent) argOutputProcessE
 	// topics[1] = active stake
 	// topics[2] = num contract users
 	// topics[3] = total contract active stake
-	// topics[4] = true - if delegator was deleted in case of withdrawal
+	// topics[4] = true - if delegator was deleted in case of withdrawal OR contract address in case of delegate from async
 	activeStake := big.NewInt(0).SetBytes(topics[1])
 
+	contractAddr := dp.pubkeyConverter.Encode(args.logAddress)
+	if len(topics) >= minNumTopicsDelegators+1 && eventIdentifierStr == delegateFunc {
+		contractAddr = dp.pubkeyConverter.Encode(topics[4])
```

Comment:
do we have a protection not to have less than 4 elements so this statement can not panic?

Reply:
we have a check in the case we have fewer topics https://github.com/ElrondNetwork/elastic-indexer-go/blob/f2d05c0e40a6dbe39bdbf81faa3deea2143e01bf/process/logsevents/delegatorsProcessor.go#L59

Reply:
ok great!